src/lxc/lxc_attach.c

```diff
@@ -273,7 +273,12 @@ static int get_pty_on_host(struct lxc_container *c, struct wrapargs *wrap, int *
 	if (lxc_console_create(conf) < 0)
 		return -1;
 	ts = conf->console.tty_state;
-	ts->stdoutfd = STDOUT_FILENO;
+	/* If stdout is not a tty but stderr is, we write to stderr. */
+	if (!isatty(STDOUT_FILENO) && isatty(STDERR_FILENO))
+		ts->stdoutfd = STDERR_FILENO;
+	else
+		ts->stdoutfd = STDOUT_FILENO;
+
```

Comment:
Oh, I see.  Martin, this is because the 'ts' only has stdin and stdout, no stderr.  The container's output is combined into the ts->stdout fd.

In your example you redirected container's stdout to /dev/null;  we need to make sure to let any valid output from container reach the console, so we send it to fd 2.